src/TLuaInterpreter.cpp

```diff
@@ -309,6 +309,11 @@ int TLuaInterpreter::raiseEvent( lua_State * L )
             pE.mArgumentList.append( QString(lua_tostring( L, i )) );
             pE.mArgumentTypeList.append( ARGUMENT_TYPE_STRING );
         }
+        else if( lua_isboolean( L, i ) )
```

Comment:
Right, casting nil or false to a boolean gives false, and anything else gives true. This change doesn't do anything with nil arguments, though - they'll still be ignored. So if you do, for example, raiseEvent("myEvent","arg1",nil,"arg3"), an event handlers registered to "myEvent" would be called as myEventHandler("myEvent","arg1","arg3"). It might be worthwhile to also add an ARGUMENT_TYPE_NIL and allow passing nil arguments. More generally, I don't like the behavior of silently omitting arguments of unrecognized types. It would seem preferable to allow as many types as possible, and have a catchall else statement that threw an error if an unsupported argument type was passed. That would be a somewhat larger change to event handling than I was looking at with this PR, which was just addressing a specific issue that was raised over on the Achaea forums, but it's something I could look into.

For now I'm inclined to leave this PR is and solicit opinions on the more general behavior separately.